src/metadata.js

```diff
@@ -1,3 +1,9 @@
+function isObject(val) {
+  return val !== null
+    && val !== undefined
+    && (typeof val === 'function' || typeof val === 'object');
```

Comment:
Just learning.

Normally I would do like this:
```js
let type = typeof val;
return val && (type === 'function' || type === 'object')
```

What are the benefits of doing `!== null`  and `!== undefined` instead of `val` && `...`